faucet.go

```diff
@@ -947,18 +886,25 @@ func (l *lightningFaucet) generateInvoice(homeTemplate *template.Template,
 func (l *lightningFaucet) payInvoice(homeTemplate *template.Template,
 	homeState *homePageContext, w http.ResponseWriter, r *http.Request) {
 
-	// Disable pay invoice if user set this parameter
-	if l.disablePayInvoices {
-		http.Error(w, "invoices payment was disabled", 403)
-		return
+	// Verify IP before open a channel
+	hostIP, err := getRealIP(r, l.cfg.UseRealIP)
+	if err != nil {
+		log.Errorf("Can't get client ip: %v", err)
```

Comment:
I don't think we should continue if we have no value for `hostIP`, it should always be possible to determine some kind of address, using `X-Real-IP`, `X-Fordwarded-For` or `RemoteAddr`. 

[testnetfauct actually panics](https://github.com/decred/testnetfaucet/blob/3f81fec5287c02a33734d50d11cbda7e640b8ac0/main.go#L89-L92) if it cannot find a client IP. This is a bit too dramatic imo. We should probably just log an error and display an error on the page.

Reply:
What kind of message?

Reply:
Just something generic like "An internal error occurred" would be fine. Or maybe a 500 or 503